### ISSUE TITLE
player:set_wield_index(n)

### ISSUE THREAD

**Issue by jordan4ibanez** [minetest/minetest#11526]
## Problem

A clear and concise description of what the problem is.
ie: Why is this needed?

To have the ability to adjust which section of the hotbar is selected. If the hotbar and inventory shrinks below the number that the current wield index is in, undefined behavior is suddenly created where a player can crash the server by pressing the drop button due to a null section of memory becoming indexed with no handling.

## Solutions

Implement set_wield_index, or automatically shrink the wield index to the greatest applicable size in the new hotbar size.

## Alternatives

Accept crashing and undefined behavoir

**Comment by sfan5** [comment]
I don't see how the existence of a bug means we should have this feature.
The bug should just be fixed on its own.

**Comment by rubenwardy** [comment]
Yeah, this feels like it should be two different issues

**Comment by SmallJoker** [comment]
This is an obscure mix of feature and bug report. Your requested feature is counter-productive on laggy servers, and has IMO too little use.

**Comment by LibraSubtilis** [comment]
I am working on a [mod pack](https://content.minetest.net/packages/LibraSubtilis/cursor/) that transforms the player into a turtle. It is possible to move and build whatever you want yourself. I also implemented a small language that enables you to automate the player turtle. I use get_wielded_item to determine the node placed by the turtle and it would be very nice to have a setter for the wield index. 
It is a small, isolated feature and I hope someone is going to implement it. 
I also wrote this on the [forum](https://forum.minetest.net/viewtopic.php?f=47&t=24455).
